frontend/templates/product/sections/ProductSection/AddToCart/index.tsx

```diff
@@ -106,19 +122,108 @@ export function AddToCart({ product, selectedVariant }: AddToCartProps) {
    );
 }
 
-function ShippingRestrictions() {
-   return (
-      <Flex py="0" fontSize="sm" align="center">
-         Shipping restrictions apply
-         <FaIcon
-            display="block"
-            icon={faCircleInfo}
-            h="4"
-            ml="1.5"
-            color="gray.400"
-         />
-      </Flex>
+type ShippingRestrictionsInfo = {
+   [key: string]: { notice: string; text: string; link?: string };
+};
+const shippingRestrictionsInfo: ShippingRestrictionsInfo = {
+   is_battery: {
+      notice: 'Shipping restrictions apply',
+      text: 'Batteries may only be shipped within the contiguous USA at this time and may only ship via standard shipping.',
+      link: 'https://help.ifixit.com/article/195-battery-shipping-warnings',
+   },
+   is_dangerous: {
+      notice: 'Shipping restrictions apply',
+      text: 'New safety regulations for dangerous goods may cause shipping delays.',
+      link: 'https://help.ifixit.com/article/264-what-are-the-shipping-restrictions-for-orders-containing-dangerous-goods',
+   },
+   us_only: {
+      notice: 'US shipping only',
+      text: 'Due to shipping regulations, iFixit may only ship this product within the United States of America.',
+   },
+   does_not_ship_to_mexico: {
+      notice: 'Mexico shipping unavailable',
+      text: 'Due to shipping regulations, iFixit may not ship this product to Mexico.',
+   },
+   fulfilled_via_dropshipping: {
+      notice: 'Partner fulfilled',
+      text: 'This product will be packed and shipped by one of our fulfillment partners.',
+   },
+};
+
+type ShippingRestrictionsProps = {
+   shippingRestrictions: string[];
+};
+
+function ShippingRestrictions({
+   shippingRestrictions,
+}: ShippingRestrictionsProps) {
+   const restrictionKeysToShow: string[] = [];
+   const primaryRestrictionKey = shippingRestrictions.find(
+      (sr) =>
+         sr !== 'fulfilled_via_dropshipping' &&
+         Object.keys(shippingRestrictionsInfo).includes(sr)
    );
+   if (primaryRestrictionKey) {
+      restrictionKeysToShow.push(primaryRestrictionKey);
+   }
+   if (shippingRestrictions?.includes('fulfilled_via_dropshipping')) {
```

Comment:
`shippingRestrictions` can't be undefined/null here right?

```suggestion
   if (shippingRestrictions.includes('fulfilled_via_dropshipping')) {
```

Reply:
Correct, I've applied the suggested change